### DISCUSSION TITLE
Tired novice programmers

### DISCUSSION THREAD

**Discussion by Amirrezaheydari81** [community/community#20126] [Other Feature Feedback, Questions, & Ideas]
What do you think beginner programmers should do so that they don't get tired during training?

**Comment by radar-blip** [comment]
While taking breaks:
Clean up code so its easy to read and follow. This makes it easier for you and others to edit at a later time (if need be).
Coffee helps too.

**Reply by Amirrezaheydari81** [reply]
It is an interesting idea, where do you think they should learn this technique? Do you think anything else can be done?

**Reply by jalanb** [reply]
They could start by finding the worst bit of code they've written. We all have at least one "It works, but ..." in our code. Fix that one.

Narrow it down to 16, then 8, 4, 2, and only 1 line of [smelly code](https://duckduckgo.com/?t=ffab&q=smelly+code&atb=v320-1&ia=web). Remove it. 

Repeat for 30 years and you'll be worth millions

**Reply by Amirrezaheydari81** [reply]
It was an interesting method :)

**Comment by mmbi18** [comment]
> What do you think beginner programmers should do so that they don't get tired during training?
In times of intellectual exhaustion,I'll start continuing the other project.
But when I can't work anymore, movies or dreams or excursions.
Now the fatigue of training time, I think you need to rest like this or go.

**Reply by Amirrezaheydari81** [reply]
It was an interesting method :)
